python_scripts/parse_kraken_to_biom.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python
+
+#-----------------------------------------------------------------------------
+# Copyright (c) 2016--, Evguenia Kopylova.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+"""
+Parse output of "kraken translate" and generate single BIOM table.
+"""
+
+import click
+import pandas as pd
+import numpy as np
+from biom.table import Table
+from biom.util import biom_open, HAVE_H5PY
+
+
+def compute_biom_table(kraken_translate_report_fp,
+                       taxonomic_rank,
+                       taxa_levels,
+                       taxa_levels_idx,
+                       columns,
+                       index):
+    """Absolute abundance of number of reads matching a defined taxa level.
+
+    Parameters
+    ----------
+    kraken_translate_report_fp: string
+        filepath to output of "kraken translate"
+    taxonomic_rank: string
+        taxonomy level (e.g., genus or species)
+    taxa_levels: dictionary
+        keys are full name taxonomic ranks and values are abbreviated ranks
+    taxa_levels_idx: dictionary
+        2-way dict storing integer depths for abbreviated taxonomic ranks
+    columns: list
+        sample IDs list
+    index: list
+        observation IDs (taxonomy strings) list
+
+    Returns
+    -------
+    biom_table: biom.Table
+        BIOM table
+    """
+    total_levels = len(taxa_levels)
+    # columns are sample IDs and rows are taxonomy strings
+    abundances = pd.DataFrame(columns=columns, index=index)    
+    taxonomic_rank_level_str = taxa_levels[taxonomic_rank]
+    taxonomic_rank_level_int = taxa_levels_idx[taxonomic_rank_level_str]
+    if taxonomic_rank_level_int < 6:
+        split_on_level = taxa_levels_idx[str(taxonomic_rank_level_int + 1)]
+    else:
+        # keep full string (to species level)
+        split_on_level = '\t'
+    with open(kraken_translate_report_fp) as kraken_translate_report_f:
+        for line in kraken_translate_report_f:
+            label, taxonomy = line.strip().split('\t')
```

Comment:
hmm, it's a 2 column file with the 2nd column containing a string without spaces/tabs, so I don't think it's necessary?